org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/util/DontAskAgainPreferences.java

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright (c) 2017 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ui.util;
+
+import org.eclipse.jface.dialogs.IDialogSettings;
+import org.eclipse.jface.dialogs.MessageDialogWithToggle;
+
+import com.google.inject.Inject;
+
+/**
+ * A utility class to remember a user's decision concerning "Don't ask again" preferences.<br>
+ * The settings are stored in {@link IDialogSettings} of your language UIPlugin.
+ *
+ * @author Titouan Vervack - Initial contribution and API
+ * @since 2.14
```

Comment:
inconsisten use of @since on class and methods. if i understood correct @since on class is enhough. correct @szarnekow

Reply:
Yes, that would be sufficient as far as I can tell. It doesn't do any harm, though

Reply:
Fixed it on methods, but left it on the constant because that's how it was in DontAskAgainDialogs.java